openapi.yaml

```diff
@@ -5053,10 +5177,13 @@ components:
       example: '2017-01-01T09:36:18Z'
     user_id:
       type: string
-      description: >-
+      description: |-
         Unique identifier of the user. MUST match the specified pattern.
-        This is usually a randomly generated internal identifier from the provider
-        not meant for displaying purposes.
+
+        For ease of use, it is NOT RECOMMENDED to use long randomly generated
+        identifiers. More readable user identifiers like `john_doe` support a
+        better user-experience as the user identifier is used in URIs for shared
+        processes, e.g. `https://example.org/api/v1.0/processes/@john_doe/my_ndvi`.
```

Comment:
I agree that `john_doe` is nicer, but this recommendation does not play well with what  OIDC gives us. For example google returns user ids like `us-a44d63b6-e090-6059-81d7-cbe7afeff6ce` and microsoft: `nIrHDS4rhk4ri738TRhtLHXdoUQ6OxZo9Ob0AS3vTig`

there is a piece of the puzzle missing here

Reply:
If there's no user name / nick name to derive such a slug from, what to use instead?

Reply:
indeed, I'm not even sure the returned userid is stable, e.g. if a user registers a new client id to work with, maybe an OIDC providers could bump/rehash the userid?

Reply:
But in general, I would expect a back-end assigns a separate user-id in addition to the external user-id. That is how I've seen it implemented.